### ISSUE TITLE
Logging train metrics on epoch doesn't work with Torchmetrics master + PL master

### ISSUE THREAD

**Issue by ethanwharris** [PyTorchLightning/pytorch-lightning#8350]
## 🐛 Bug

Logging train metrics on epoch doesn't work with Torchmetrics master + PL master

I get this warning:
```
 UserWarning: The ``compute`` method of metric Accuracy was called before the ``update`` method which may lead to errors, as metric states have not yet been updated.
```

### To Reproduce

https://colab.research.google.com/drive/1zn7A3BpvYRmxegIJxsWVQbHBhMsj5mTl?usp=sharing

### Expected behavior

On epoch metrics should be logged correctly

**Comment by ethanwharris** [comment]
Closing this as it's down to user error. You need to use separate metric objects for each phase.
